test/src/test/java/hudson/model/NodeTest.java

```diff
@@ -237,7 +242,95 @@ public void testToComputer() throws Exception {
+        node.setLabelString("label1 label2");
+        MavenModuleSet mavenProject = j.createMavenProject();
+        mavenProject.setAssignedLabel(j.jenkins.getLabel("label1"));
+        RunLoadCounter.prepare(mavenProject);
+        j.assertBuildStatus(Result.FAILURE, mavenProject.scheduleBuild2(0).get());
+        Integer labelCount = RunLoadCounter.assertMaxLoads(mavenProject, 0, new Callable<Integer>() {
+            @Override
+            public Integer call() throws Exception {
+                return j.jenkins.getLabel("label1").getTiedJobCount();
+            }
+        });
+
+        assertEquals("Should have only one job tied to label.",
+                1, labelCount.intValue());
+    }
     
+    /**
+     * Create two projects which have the same label and verify that both are accounted for when getting a count
+     * of the jobs tied to the current label.
+     *
+     */
+    @Issue("JENKINS-26391")
+    @Test
+    public void testGetAssignedLabelMultipleSlaves() throws Exception {
+        final Node node1 = j.createOnlineSlave();
+        node1.setLabelString("label1");
+        final Node node2 = j.createOnlineSlave();
+        node1.setLabelString("label1");
+
+        MavenModuleSet project = j.createMavenProject();
+        project.setAssignedLabel(j.jenkins.getLabel("label1"));
+        j.assertBuildStatus(Result.FAILURE, project.scheduleBuild2(0).get());
+
+        MavenModuleSet project2 = j.createMavenProject();
+        project2.setAssignedLabel(j.jenkins.getLabel("label1"));
+        j.assertBuildStatus(Result.FAILURE, project2.scheduleBuild2(0).get());
+
+        assertEquals("Two jobs should be tied to this label.",
+                2, j.jenkins.getLabel("label1").getTiedJobCount());
+    }
+
+    /**
+     * Verify that when a label is removed from a job that the tied job count does not include the removed job.
+     */
+    @Issue("JENKINS-26391")
+    @Test
+    public void testGetAssignedLabelWhenLabelRemoveFromProject() throws Exception {
+        final Node node = j.createOnlineSlave();
+        node.setLabelString("label1");
+
+        MavenModuleSet project = j.createMavenProject();
+        project.setAssignedLabel(j.jenkins.getLabel("label1"));
+        j.assertBuildStatus(Result.FAILURE, project.scheduleBuild2(0).get());
+
+        project.setAssignedLabel(null);
+        assertEquals("Label1 should have no tied jobs after the job label was removed.",
+                0, j.jenkins.getLabel("label1").getTiedJobCount());
+    }
+
+    /**
+     * Create a project with the OR label expression.
+     */
+    @Issue("JENKINS-26391")
+    @Test
+    public void testGetAssignedLabelWithLabelExpression() throws Exception {
+        Node node = j.createOnlineSlave();
+        node.setLabelString("label1 label2");
+
+        FreeStyleProject project = j.createFreeStyleProject();
+        project.setAssignedLabel(new LabelExpression.Or(j.jenkins.getLabel("label1"), j.jenkins.getLabel("label2")));
```

Comment:
there is no negative testing here... what about a project that has `label1 && (label2 || label3)` or even a project that has `label1 && label3` or `label3 label4`